test/UnitTests/MSTest.Analyzers.UnitTests/UseNullableForIsNullAndIsNotNullAssertionsAnalyzerTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using VerifyCS = MSTest.Analyzers.Test.CSharpCodeFixVerifier<
+    MSTest.Analyzers.UseNullableForIsNullAndIsNotNullAssertionsAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace MSTest.Analyzers.Test;
+
+[TestGroup]
+public sealed class UseNullableForIsNullAndIsNotNullAssertionsAnalyzerTests(ITestExecutionContext testExecutionContext) : TestBase(testExecutionContext)
+{
+    public async Task WhenIsNullAssertion_ValueParameterIsNullable_NoDiagnostic()
+    {
+        string code = """
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+            [TestClass]
+            public class MyTestClass
+            {
+                [TestMethod]
+                public void MyTestMethod()
+                {
+                    string? var = null;
```

Comment:
Since `string` is a reference type, this will not result in `Nullable<string>`. `Nullable<T>` can only be used with value types. To be sure, maybe it is better to actually write `Nullable<...>` in the tests, instead of the short-hand `?`.